core/state/reader.go

```diff
@@ -207,16 +215,39 @@ type trieReader struct {
 
 // trieReader constructs a trie reader of the specific state. An error will be
 // returned if the associated trie specified by root is not existent.
-func newTrieReader(root common.Hash, db *triedb.Database, cache *utils.PointCache) (*trieReader, error) {
+func newTrieReader(root common.Hash, db, verkledb *triedb.Database, cache *utils.PointCache) (*trieReader, error) {
 	var (
-		tr  Trie
-		err error
+		tr   Trie
+		mptr *trie.StateTrie
+		vktr *trie.VerkleTrie
+		err  error
 	)
-	if !db.IsVerkle() {
-		tr, err = trie.NewStateTrie(trie.StateTrieID(root), db)
+	// get the transition status from the MPT triedb
+	ts, err := db.LoadTransitionState(root)
```

Comment:
note that this method doesn't currently exist, I'm proposing that we move it from `(*state.Database).LoadTransitionState`) and keep it inside the tree somehow.